concurrency-limiter/concurrency_limiter.go

```diff
@@ -55,13 +61,13 @@ func (cl *ConcurrencyLimiter) ServeHTTP(w http.ResponseWriter, r *http.Request)
 // NewConcurrencyLimiter creates a handler which limits the active number of active, concurrent
 // requests. If the concurrency limit is less than, or equal to 0, then it will just return the handler
```

Comment:
This doc is no longer true, it only returns the ConcurrencyLimiter. 

Also, it seems like a but might be introduced when `concurrencyLimit=0`. It is now possible that `cl.maxInflightRequests == 0`. So, on line 50 https://github.com/openfaas/faas-middleware/pull/3/files#diff-68c9ecc3bc5f7baf7d78b948ccdf006b53a15560820a3f958bf58a72ab22d65cR50 a limit of 0 will always be met, right? 

It seems that line 50 needs to be updated, add something like

```go 
if cl.maxInflightRequests == 0 {
   cl.bacendHTTPHandler.ServeHTTP(w,r)
   return
}
```

Or the docs should be updated to indicate that when the value is 0, all requests will be rejected.

Reply:
That makes sense, thanks for picking that up.